### PR TITLE
Trajectory protobuf

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: physics/trajectory.hpp [hunk 1]
```diff
@@ -162,10 +162,13 @@ class Trajectory {
...
-  // This trajectory must be a root.
+  // This trajectory must be a root.  The intrinsic acceleration is not
+  // serialized.  The body is not owned, and therefore is not serialized.
   void WriteToMessage(not_null<serialization::Trajectory*> const message) const;
-  // TODO(egg): implement when |Body| is serializable.
-  static Trajectory ReadFromMessage(serialization::Trajectory const& message);
+
+  static not_null<std::unique_ptr<Trajectory>> ReadFromMessage(
+      serialization::Trajectory const& message,
+      not_null<Body const*> body);
```
Comment (pleroy) [inline physics/trajectory.hpp]: const

File: physics/trajectory.hpp [hunk 2]
```diff
@@ -162,10 +162,13 @@ class Trajectory {
...
       Instant const& time) const;
 
-  // This trajectory must be a root.
+  // This trajectory must be a root.  The intrinsic acceleration is not
+  // serialized.  The body is not owned, and therefore is not serialized.
   void WriteToMessage(not_null<serialization::Trajectory*> const message) const;
-  // TODO(egg): implement when |Body| is serializable.
-  static Trajectory ReadFromMessage(serialization::Trajectory const& message);
+
+  static not_null<std::unique_ptr<Trajectory>> ReadFromMessage(
```
Comment (pleroy) [inline physics/trajectory.hpp]: Why does it return a pointer and not a value?

File: physics/trajectory_body.hpp [hunk 3]
```diff
@@ -272,6 +272,15 @@ void Trajectory<Frame>::WriteToMessage(
 }
 
 template<typename Frame>
+not_null<std::unique_ptr<Trajectory<Frame>>> Trajectory<Frame>::ReadFromMessage(
+    serialization::Trajectory const& message,
+    not_null<Body const*> const body) {
+  auto result = make_not_null_unique<Trajectory>(body);
```
Comment (pleroy) [inline physics/trajectory_body.hpp]: s/result/trajectory/

File: physics/trajectory_body.hpp [hunk 4]
```diff
@@ -404,5 +413,30 @@ void Trajectory<Frame>::WriteSubTreeToMessage(
   }
 }
 
+template<typename Frame>
+void Trajectory<Frame>::FillSubTreeFromMessage(
+    serialization::Trajectory const& message) {
+  auto timeline_iterator = message.timeline().begin();
```
Comment (pleroy) [inline physics/trajectory_body.hpp]: s/timeline_iterator/it/ maybe?

File: physics/trajectory.hpp (hunk 2 above)
Comment (eggrobin) [inline physics/trajectory.hpp]: #410

File: physics/trajectory.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/trajectory.hpp]: Done.

File: physics/trajectory_body.hpp (hunk 3 above)
Comment (eggrobin) [inline physics/trajectory_body.hpp]: Done.

File: physics/trajectory_body.hpp (hunk 4 above)
Comment (eggrobin) [inline physics/trajectory_body.hpp]: `timeline_it`, with these nested loops I want to know what this iterates over.

**Pull request merged**